caikit/runtime/service_factory.py

```diff
@@ -203,9 +206,23 @@ def _get_and_filter_modules(
 
         for ck_module in modules:
             # Only create for modules from defined included and exclusion list
-            module_info = get_module_info(ck_module)
-            if excluded_task_types and module_info.type in excluded_task_types:
-                log.debug("Skipping module %s of type %s", ck_module, module_info.type)
+            # module_info = get_module_info(ck_module) # Not used
+
+            # --------------Changes added ----------------------
+            if not ck_module.TASK_CLASS:
+                continue
+            module_info_task = None
+            if ck_module.TASK_CLASS:
+                module_info_task = ck_module.TASK_CLASS.__name__
+                module_info_task = re.sub(
+                    r"(?<!^)(?=[A-Z])", "_", module_info_task
+                ).lower()
```

Comment:
Nice! I appreciate the dedication to keeping the same configuration api working.

But, this PR is actually going to be a breaking change anyway because we're no longer generating inference RPCs for modules with no task explicitly defined. So I think we might want to go ahead and simplify this config to match the actual name of the task class. 

```
if excluded_task_types and ck_module.TASK_CLASS.__name__ in excluded_task_types:
```
should work just fine, right?

Reply:
I think this would not work due to one small thing. I will discuss with you.